packages/build-info/src/detect-build-system.ts

```diff
@@ -0,0 +1,145 @@
+import { readFileSync } from 'fs'
+import path from 'path'
+
+import { findUpSync } from 'find-up'
+import { PackageJson } from 'read-pkg'
+
+export type BuildSystem = {
+  name: string
+  version?: string | undefined
+}
+
+export const detectBuildSystems = async (baseDir: string, rootDir?: string): Promise<BuildSystem[]> => {
+  const buildTools = Object.keys(BUILD_SYSTEMS)
+  const buildSystems = await Promise.all(buildTools.map(async (tool) => await BUILD_SYSTEMS[tool](baseDir, rootDir)))
+  return buildSystems.filter((tool) => tool != undefined)
+}
+
+const BUILD_SYSTEMS = {
```

Comment:
NIT: I would use type inference on the functions by typing the object.

```typescript
type BuildSystem = (baseDir: string, rootDir?: string) => Promise<BuildSystem | undefined>;

const BUILD_SYSTEMS: Record<string, BuildSystem> = {
}
```

Then you don't have to type all the functions